esmvalcore/cmor/_fixes/icon/icon.py

```diff
@@ -274,42 +280,197 @@ def _fix_time(self, cube, cubes):
 
         return cube
 
+    def _get_mesh(self, cube):
+        """Create mesh from horizontal grid file.
+
+        Note
+        ----
+        This functions creates a new :class:`iris.experimental.ugrid.Mesh` from
+        the ``clat`` (already present in the cube), ``clon`` (already present
+        in the cube), ``vertex_index``, ``vertex_of_cell``, ``vlat``, and
+        ``vlon`` variables of the horizontal grid file.
+
+        We do not use :func:`iris.experimental.ugrid.Mesh.from_coords` with the
+        existing latitude and longitude coordinates here because this would
+        produce lots of duplicated entries for the node coordinates. The reason
+        for this is that the node coordinates are constructed from the bounds;
+        since each node is contained 6 times in the bounds array (each node is
+        shared by 6 neighboring cells) the number of nodes is 6 times higher
+        with :func:`iris.experimental.ugrid.Mesh.from_coords` compared to using
+        the information already present in the horizontal grid file.
```

Comment:
Is this something that could be improved in iris?

Reply:
Definitely! The current approach is the most straightforward and easiest one (just collect nodes from the bounds, regardless of duplicates), but it's certainly not super hard to come up with a solution that handles duplicates correctly (note that you would also need to adapt the connectivity accordingly).

However, for ICON all this information is already provided in the grid file. There is no need to calculate it again. Thus, I think we should stick to the approach used here, even if iris updates `iris.experimental.ugrid.Mesh.from_coords` tomorrow.

Reply:
>  it's certainly not super hard to come up with a solution that handles duplicates correctly (note that you would also need to adapt the connectivity accordingly).

Is there already an iris issue open about this? If not, it would be nice to open one and explain how things could be improved.

> However, for ICON all this information is already provided in the grid file. There is no need to calculate it again. Thus, I think we should stick to the approach used here, even if iris updates iris.experimental.ugrid.Mesh.from_coords tomorrow.

:+1:

Reply:
They are aware of it (see note [here](https://scitools-iris.readthedocs.io/en/latest/generated/api/iris/experimental/ugrid.html?highlight=from_coords#iris.experimental.ugrid.Mesh.from_coords)), so I don't think it's necessary to open an issue :+1: